File: src/Python/easy/string/problem1684.py
from typing import List


class Solution:
    def countConsistentStrings(self, allowed: str, words: List[str]) -> int:
        i = 0
        flag = None
        for idx, word in enumerate(words):
            for w in word:
                if w not in allowed:
                    if flag == idx:
                        continue
                    else:
                        i += 1
                        flag = idx
        return len(words) - i





allowed = "ab" 
words = ["ad","bd","aaab","baa","badab"]


# allowed = "cad"
# words = ["cc","acd","b","ba","bac","bad","ac","d"]

# allowed = "fstqyienx" 
# words = ["n","eeitfns","eqqqsfs","i","feniqis","lhoa","yqyitei","sqtn","kug","z","neqqis"]


solution = Solution()
solution.countConsistentStrings(allowed, words)

File: src/JavaScript/easy/string/problem1684.js
/**
 * @param {string} allowed
 * @param {string[]} words
 * @return {number}
 */
var countConsistentStrings = function(allowed, words) {
    let a = 0
    label : for(let i = 0; i < words.length; i++){
        for(let j = 0; j < words[i].split("").length; j++){
            if(!allowed.includes(words[i][j])){
                a++
                continue label;
            }
        }
    }
    return words.length - a
};

// let allowed = "ab", words = ["ad","bd","aaab","baa","badab"]
// let allowed = "abc", words = ["a","b","c","ab","ac","bc","abc"]
// let allowed = "cad", words = ["cc","acd","b","ba","bac","bad","ac","d"]
let allowed = "fstqyienx", words = ["n","eeitfns","eqqqsfs","i","feniqis","lhoa","yqyitei","sqtn","kug","z","neqqis"]

countConsistentStrings(allowed, words)